### PR TITLE
Pptp 2377

### PR DESCRIPTION
**Pull request by ajchilts**
### Description of Work carried through

Currently the returns front end does the calculation builds a `TaxReturn` object and passes it to the back end for submission. The back end then builds a number of objects and repeats the calculation before finally submitting to ETMP. 

As the UserAnswers cache now lies within the backend we can refractor the code such that the calculation lives in the back end only and the ETP submission is build from the UserAnswers without the need to construct pointless objects. Validation can also be included at this level to prevent invalid submissions. 

Raise bug with Etmp, their API should have rejected the duplicate according to the spec
https://github.com/hmrc/plastic-packaging-tax-stub/blob/main/conf/api-docs/api-1713-create-update-ppt-return-1.3.0.yaml

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/ReturnsSubmissionRequest.scala [hunk 1]
```diff
@@ -39,30 +42,37 @@ case class EisReturnDetails(
 object EisReturnDetails {
   implicit val format: OFormat[EisReturnDetails] = Json.format[EisReturnDetails]
 
-  def apply(taxReturn: TaxReturn, taxRatePoundsPerKg: BigDecimal): EisReturnDetails = {
-    val manufacturedWeightKg = taxReturn.manufacturedPlasticWeight.totalKg
-    val importedWeightKg     = taxReturn.importedPlasticWeight.totalKg
-    val liableKg             = manufacturedWeightKg + importedWeightKg
+  def apply(returnValues: ReturnValues, calculations: Calculations): EisReturnDetails = {
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/ReturnsSubmissionRequest.scala:45]: shall we make return values a trait, with a _amend_ and _new_ flavour 

would save on duplication

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/ReturnsSubmissionRequest.scala [hunk 2]
```diff
@@ -80,16 +90,26 @@ object ReturnsSubmissionRequest {
 
   implicit val format: OFormat[ReturnsSubmissionRequest] = Json.format[ReturnsSubmissionRequest]
 
-  def apply(taxReturn: TaxReturn, taxRatePoundsPerKg: BigDecimal, submissionId: Option[String]): ReturnsSubmissionRequest = {
+  def apply(returnValues: ReturnValues, calculations: Calculations, submissionId: Option[String], returnType: ReturnType): ReturnsSubmissionRequest = {
 
-    if (taxReturn.returnType == (ReturnType.AMEND) && submissionId.isEmpty)
+    if (returnType == (ReturnType.AMEND) && submissionId.isEmpty)
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/connectors/models/eis/returns/ReturnsSubmissionRequest.scala:95]: this cannot be right 🤔

File: app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/gettables/amends/AmendRecycledPlasticPackagingGettable.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.plasticpackagingtaxreturns.models.cache.gettables.amends
+
+import play.api.libs.json.JsPath
+import uk.gov.hmrc.plasticpackagingtaxreturns.models.cache.Gettable
+
+case object AmendRecycledPlasticPackagingPage extends Gettable[Long] {
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxreturns/models/cache/gettables/amends/AmendRecycledPlasticPackagingGettable.scala:22]: this is a Gettable not a Page

File: test/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/controllers/CalculationsControllerSpec.scala [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      |        "obligation" : {
+      |            "periodKey" : "21C4"
+      |        },
+      |        "manufacturedPlasticPackagingWeight" : 100,
+      |        "importedPlasticPackagingWeight" : 0,
+      |        "exportedPlasticPackagingWeight" : 0,
+      |        "nonExportedHumanMedicinesPlasticPackagingWeight" : 10,
+      |        "nonExportRecycledPlasticPackagingWeight" : 5
+      |    }""".stripMargin).asInstanceOf[JsObject]
+
```
Comment (lmcewanhmrc) [inline test/uk/gov/hmrc/plasticpackagingtaxreturns/controllers/controllers/CalculationsControllerSpec.scala:64]: Nice ⬆️

**Pull request merged**